src/components/calcite-input/calcite-input.tsx

```diff
@@ -502,19 +502,24 @@ export class CalciteInput {
     if (this.type !== "number") {
       return;
     }
-    const decimals = this.value?.split(".")[1]?.length || 0;
+    const decimals = this.step?.toString().split(".")[1]?.length || 0;
     const inputMax = this.maxString ? parseFloat(this.maxString) : null;
     const inputMin = this.minString ? parseFloat(this.minString) : null;
     const inputStep = this.step === "any" ? 1 : Math.abs(this.step || 1);
-    let inputVal = this.value && this.value !== "" ? parseFloat(this.value) : 0;
+    const inputVal =
+      this.value && this.value !== ""
```

Comment:
Nitpick: `this.value` is referenced a bit, so can you store it in a local variable?

Reply:
This line can be simplified to: `const inputVal = this.value ? ...`

```js
// when "1" => ("1") && ("1" !== "" /* will never be true */);
// when null/undefined => (null/undefined) && (/* not reached */ null/undefined !== "");
// when "" => ("") && (/* not reached */ null !== "");
```